### PR TITLE
Way to pass styles to nested view

### PR DESCRIPTION
**Pull request by slavikmanukyan**
Ability to pass styles to nested view for _position_ behavior.

Use case:

``` jsx
<KeyboardAvoidingView
          behavior={'position'}
          style={{ flex: 1 }}
          contentContainerStyle={{ flex: 1 }}
          keyboardVerticalOffset={ - VERTICAL_OFFSET }
>
          <SomeContainer   />
</KeyboardAvoidingView>
```

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
By analyzing the blame information on this pull request, we identified @nicklockwood and @davidaurelio to be potential reviewers.

**Comment by facebook-github-bot-2** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-1** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by ide** [comment]
Can you rename this to contentContainerStyle?

**Comment by facebook-github-bot-2** [comment]
@slavikmanukyan updated the pull request.

**Comment by facebook-github-bot-4** [comment]
@slavikmanukyan updated the pull request.

**Comment by janicduplessis** [comment]
@slavikmanukyan Can you also add a docblock for the prop that explain what it does, make sure to mention that it only applies to the position behavior.

**Comment by satya164** [comment]
Maybe we should add a custom propType here and warn if it is passed when behaviour is not position.

**Comment by facebook-github-bot-8** [comment]
@slavikmanukyan updated the pull request.

**Comment by facebook-github-bot-8** [comment]
@slavikmanukyan updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@slavikmanukyan updated the pull request.

File: Libraries/Components/Keyboard/KeyboardAvoidingView.js [hunk 1]
```diff
@@ -160,9 +173,11 @@ const KeyboardAvoidingView = React.createClass({
 
       case 'position':
         const positionStyle = {bottom: this.state.bottom};
+        const { contentContainerStyle } = this.props;
+
         return (
           <View ref={viewRef} style={style} onLayout={this.onLayout} {...props}>
-            <View style={positionStyle}>
+            <View style={[positionStyle, contentContainerStyle]}>
```
Comment (javache) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: Do `[contentContainerStyle, positionStyle]` so position overrides anything set previously.

File: Libraries/Components/Keyboard/KeyboardAvoidingView.js [hunk 2]
```diff
@@ -53,7 +53,20 @@ const KeyboardAvoidingView = React.createClass({
...
     ...View.propTypes,
     behavior: PropTypes.oneOf(['height', 'position', 'padding']),
+    
+    /**
+     * The style of the content container(View) when behavior is 'position'.
+     */
+    contentContainerStyle (props, propName, componentName) {
+      const isStyle = View.propTypes.style;
 
+      if (props[propName] && props.behavior !== 'position') {
```
Comment (javache) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: I wouldn't bother with this, it doesn't work with Stylesheets or with an array of styles. Just use View.propTypes.style for the whole thing.

File: Libraries/Components/Keyboard/KeyboardAvoidingView.js (hunk 1 above)
Comment (satya164) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: @javache Isn't it better to allow overriding position? If someone doesn't want to override position, he won't pass a prop.

Use case: say there's a overlayed view, in which case you only want to adjust the bottom in the overlayed view, in the underlying view, you can disable the behaviour by passing `bottom: 0`

**Comment by facebook-github-bot-5** [comment]
@slavikmanukyan updated the pull request.

**Comment by javache** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-5** [comment]
Thanks for importing.

**Pull request closed without merging**